active-rfcs/0000-finalize-head-body.md

```diff
@@ -0,0 +1,65 @@
+- Start Date: 11/30/2021
+- Reference Issues: https://github.com/withastro/rfcs/discussions/15
+- Implementation PR:
+
+# Summary
+
+Finalize behavior around `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` elements in Astro component templates.
+
+# Motivation
+
+@jonathantneal and I were chatting about [RFC: Finalize v1.0 style and script API for components](https://github.com/withastro/rfcs/pull/12) and discussing `<head>` behavior & bugs in Astro. We came to the realization that only supporting `<head>` in your top-level pages (and not in components) doesn't make sense for anyone using [layout components](https://docs.astro.build/core-concepts/layouts/), which can act as a component but often defines the entire `<head>`. This is a very common use-case for Astro users.
+
+```astro
+---
+// src/pages/index.astro
+import CommonLayout from '../layouts/CommonLayout.astro';
+---
+<!-- this is valid, where CommonLayout contains `<head>`, `<body>`, etc.
+<CommonLayout someProp="..." />
+```
+
+This was something that we had supported in v0.20, which was accidentally removed as a part of the new compiler. Adding back this behavior could be considered a regression bug fix and not necessarily requiring an RFC (and I think Nate is already working on a fix for this to add back previous behavior), but this feels like a good time to commit to a larger plan around `<head>` and other similar HTML elements.
```

Comment:
I think having these paragraphs makes the rest of the RFC confusing, because as you say here, the motivation for an API change wouldn't actually be because of a regression, would it? 

If so the motivation becomes irrelevant as soon as the regression is fixed. I think this motivation section should contain the movations/use-cases assuming that this bug didn't exist.

Reply:
This was meant to give @natemoo-re the freedom to keep working on a smaller fix for `<head>` layout behavior as a short-term workaround for everyone blocked by the `v0.20 -> v0.21` behavior change, without feeling blocked by this RFC.

Agreed that this paragraph can be removed if it's distracting. The larger RFC is attempting to make a behavior change explicit across head, body, and `<html>` so that we can document, test, and then commit to it. This is different from what we currently support in v0.21 today which is much less defined (as far as I can tell).